sync/sync_head.go

```diff
@@ -25,26 +27,24 @@ func (s *Syncer[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, err
 		return sbjHead, nil
 	}
 	// otherwise, request head from the network
-	// TODO: Besides requesting we should listen for new gossiped headers and cancel request if so
-	//
-	// single-flight protection
-	// ensure only one Head is requested at the time
+	s.metrics.outdatedHead(s.ctx)
```

Comment:
This should be below the lock. If we get multiple simultaneous routines here asking for the head, we won't benefit from reporting each routine. What we need to know is if there was such an event at the point of time.